### PR TITLE
dev-libs/yaz-5.26.1 version bump

### PR DESCRIPTION
**Pull request by gjaekel**
based on yaz-3.0.53.ebuild

I'll also try to contribute an ebuild for the  YAZ++-API as   dev-libs/yazpp  and the YAZProxy as  net-misc/yazproxy

Bug: https://bugs.gentoo.org/637404
Signed-off-by: Guido Jäkel G.Jaekel@DNB.DE

### CODE REVIEW

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Copyright 1999-2016 Gentoo Foundation
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: your calendar is obsolete

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+EAPI=5
+inherit eutils autotools
+
+DESCRIPTION="C toolkit for Z39.50v3 clients and servers"
+HOMEPAGE="http://www.indexdata.dk/yaz"
+SRC_URI="http://ftp.indexdata.dk/pub/${PN}/${P}.tar.gz"
+
+LICENSE="BSD GPL-2"
+SLOT="0"
+KEYWORDS="alpha amd64 arm hppa ~ia64 ~mips ~ppc ppc64 ~s390 ~sh ~sparc x86"
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: do not commit straight to stable for newer versions

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+src_configure() {
+        econf \
+                --enable-static \
+                --enable-shared \
+                $(use_enable debug memdebug) \
+                $(use_enable icu) \
+                $(use_enable tcpd tcpd /usr)
+}
+
+src_compile() {
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: it is default behaviour

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        icu? ( dev-libs/icu:= )
+        tcpd? ( sys-apps/tcp-wrappers )
+        ziffy? ( net-libs/libpcap )"
+DEPEND="${RDEPEND}
+        virtual/pkgconfig
+        dev-lang/tcl:0
+        >=sys-devel/libtool-2"
+
+src_prepare() {
+        epatch "${FILESDIR}"/${PN}-5.26.1-icu-automagic.patch
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: default and eapply

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+DESCRIPTION="C toolkit for Z39.50v3 clients and servers"
+HOMEPAGE="http://www.indexdata.dk/yaz"
+SRC_URI="http://ftp.indexdata.dk/pub/${PN}/${P}.tar.gz"
+
+LICENSE="BSD GPL-2"
+SLOT="0"
+KEYWORDS="alpha amd64 arm hppa ~ia64 ~mips ~ppc ppc64 ~s390 ~sh ~sparc x86"
+IUSE="debug icu libressl tcpd ziffy"
+
+RDEPEND="dev-libs/libxml2
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: slot it

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+src_install() {
+        local docdir="/usr/share/doc/${PF}"
+        emake DESTDIR="${D}" docdir="${docdir}" install || die "install failed"
+
+        dodir ${docdir}/html
+        mv -f "${D}"/${docdir}/*.{html,png} "${D}"/${docdir}/html/ || die "Failed to move HTML docs"
+        mv -f "${D}"/usr/share/doc/${PN}/common "${D}"/${docdir}/html/ || die "Failed to move HTML docs"
+        rm -rf "${D}"/usr/share/doc/${PN}
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: `|| die`

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        >=sys-devel/libtool-2"
+
+src_prepare() {
+        epatch "${FILESDIR}"/${PN}-5.26.1-icu-automagic.patch
+        AT_M4DIR="m4" eautoreconf
+}
+
+src_configure() {
+        econf \
+                --enable-static \
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: how is that? you either enable static or shared, nobody wants static libs, so either bind it to the useflag or disable at all

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,56 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
+inherit eutils autotools
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: you do not need eutils here

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,56 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: Do not use obsolete EAPI please

File: dev-libs/yaz/yaz-5.26.1.ebuild [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+inherit eutils autotools
+
+DESCRIPTION="C toolkit for Z39.50v3 clients and servers"
+HOMEPAGE="http://www.indexdata.dk/yaz"
+SRC_URI="http://ftp.indexdata.dk/pub/${PN}/${P}.tar.gz"
+
+LICENSE="BSD GPL-2"
+SLOT="0"
+KEYWORDS="alpha amd64 arm hppa ~ia64 ~mips ~ppc ppc64 ~s390 ~sh ~sparc x86"
+IUSE="debug icu libressl tcpd ziffy"
```
Comment (Zlogene) [inline dev-libs/yaz/yaz-5.26.1.ebuild]: The `ziffy` use flage only pulls a dep, please drop it and pull the dep unconditionally

**Comment by Zlogene** [comment]
Also please squash your commits into the single one

**Comment by gjaekel** [comment]
Dear Mikle,
I'm sorry, I just try to do a version bump based on the existing ebuild. I'm no Gentoo Portage developer, I'm not familiar with it in the deep. I have some background but I'm not working as developer at all. I'm able to read and understand code in may languages, but not to create it from the scatch in most cases.

I understand you comments but have no experience to solve it in a proper way. Maybe you may help me with this.

Thank you

**Comment by a17r** [comment]
@robbat2 anything going to happen with this?

**Comment by gjaekel** [comment]
Dear Andreas,

I'll ask my coworker if he's still interested in to get YAZ suite running on Gentoo for operational purposes. Maybe he is able to give me some input for what this is meant to use for.

But may you also help with this?

**Comment by a17r** [comment]
> But may you also help with this?

For the moment I'll assume that @robbat2 self-assigned this package for a reason.

There are already existing review comments, if you have trouble following them please respond directly on the line comments in question.

**Comment by robbat2** [comment]
I'm watching it because I have used Yaz in the past. But somebody needs to fix the issues with the PR or explicitly state that it's just a copy of the existing ebuild, and the cleanup should be in a new revision.

**Comment by mgorny** [comment]
Closing due to no progress.

**Comment by mgorny** [comment]
(please reopen once you start addressing the problems)

**Pull request closed without merging**